### PR TITLE
fix: `Sentinel` rendering and wrong PLP's pagination behavior (#2585)

### PR DESCRIPTION
**Pull request by lucasfp13**
## What's the purpose of this pull request?

This PR intends to fix the `Sentinel` behavior by turn it a wrapper for the `ProductGrid` component, also preventing the loop of requests happening when the user load more pages (by clicking on the `Load more products` button).

## How it works?

Currently, `Sentinel` is being rendered as a side `<div />` with the `ProductGrid` component, disabling us to control the visibility of viewport elements properly.

*Before*
**(The red line is the `Sentinel`)**

Screenshot 2024-12-05 at 13 07 20

With these changes, the `Sentinel` will wrap the `ProductGrid`, thus having the same height, enabling the use of `threshold` option (see the API options [`here`](https://github.com/thebuilder/react-intersection-observer?tab=readme-ov-file#options)) to trigger the next page (and replace the `page` query parameter) only when necessary.

*After*

Screenshot 2024-12-05 at 13 34 49

## How to test it?

Test the `/office` page with different sizes to see if the loop of requests still happens and if the `page` query parameter is still being updated correctly.

### Starters Deploy Preview

vtex-sites/starter.store#630

## References

- Using the `useInView` hook: https://github.com/thebuilder/react-intersection-observer

### CODE REVIEW

File: packages/core/src/sdk/search/Sentinel.tsx [hunk 1]
```diff
@@ -38,19 +39,24 @@ const replacePagination = (page: string, router: NextRouter) => {
...
+  pageSize,
+  products,
+  title,
+  children,
+}: PropsWithChildren<SentinelProps>) {
   const router = useRouter()
-
+  const { pages } = useSearch()
+  const { isDesktop } = useScreenResize()
+  const { ref, inView } = useInView({ threshold: isDesktop ? 0.7 : 0.3 })
```
Comment (lucasfp13) [inline packages/core/src/sdk/search/Sentinel.tsx:52]: The values here were based on my testing.
The mobile version seems fine with 30%. Desktop is working well with 70%, but other screen sizes can behave differently.

**Review by eduardoformiga — APPROVED** [review]
LGTM!

**Review by lariciamota — APPROVED** [review]

File: packages/core/src/sdk/search/Sentinel.tsx [hunk 2]
```diff
@@ -48,8 +48,8 @@ function Sentinel({
 }: PropsWithChildren<SentinelProps>) {
   const router = useRouter()
   const { pages } = useSearch()
-  const { isDesktop } = useScreenResize()
-  const { ref, inView } = useInView({ threshold: isDesktop ? 0.7 : 0.3 })
+  const { isMobile } = useScreenResize()
+  const { ref, inView } = useInView({ threshold: isMobile ? 0.3 : 0.7 })
```
Comment (hellofanny) [inline packages/core/src/sdk/search/Sentinel.tsx:52]: 💯

**Review by hellofanny — APPROVED** [review]
Niceee! 🚀 

After the last [change](https://github.com/vtex/faststore/pull/2585/commits/dfe6ccc040663afa7ff3068f813965a273832ef8), I've re-tried on some "problematic" devices `ipad mini (768 x 1024)`, `ipad air (820 x 1180)` and `ipad pro (1024 x 1366)` and its working fine (acceptable threshold values) ✅

**Pull request merged**